nano/node/bootstrap/bootstrap.cpp

```diff
@@ -617,6 +617,7 @@ void nano::bootstrap_attempt::attempt_restart_check (nano::unique_lock<std::mute
 	if (!frontiers_confirmed && (requeued_pulls > (!node->network_params.network.is_test_network () ? nano::bootstrap_limits::requeued_pulls_limit : nano::bootstrap_limits::requeued_pulls_limit_test) || total_blocks > nano::bootstrap_limits::frontier_confirmation_blocks_limit))
 	{
 		confirm_frontiers (lock_a);
+		assert (!lock_a.try_lock ());
```

Comment:
"If try_lock is called by a thread that already owns the mutex, the behavior is undefined." https://en.cppreference.com/w/cpp/thread/mutex/try_lock
Can that happen here?

Reply:
Interesting. No such mention in https://en.cppreference.com/w/cpp/thread/unique_lock/try_lock (for unique locks), although we use this idiom at least 10 times throughout the code directly with mutexes.

Reply:
It also says it effectively just calls `mutex.try_lock`. I guess they are all surrounded in `assert`s so no user would see anything anyway, so it's probably fine.

Reply:
Do you think it's better to move it outside the assert and use `owns_lock()` ?

Reply:
`owns_lock` is probably better yes